### PR TITLE
use versioned origin image for integration tests

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/image-registry/issues/51

### CODE REVIEW

**Comment by bparees** [comment]
@stevekuznetsov i'm interested in your thoughts on this and how it could/should be done differently to be more universally applicable across repositories/test jobs.

File: hack/test-integration.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
+
+ORIGIN_VERSION=latest
+if [[ -n "$PULL_BASE_REF" && $PULL_BASE_REF != "master" ]]; then
```
Comment (stevekuznetsov) [inline hack/test-integration.sh]: eek hate to leak ci into repos...

Comment (bparees) [reply hack/test-integration.sh]: @stevekuznetsov sure. i'm open to suggestions......

Comment (stevekuznetsov) [reply hack/test-integration.sh]: this logic in aos-cd-jobs?

Comment (stevekuznetsov) [reply hack/test-integration.sh]: the script in the repo takes as input the version it needs?

File: pkg/testframework/versions.go [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+	"os"
+)
+
+var (
+	// originImageRef is the Docker image name of the OpenShift Origin container.
+	originImageRef = "docker.io/openshift/origin:latest"
+)
+
+func init() {
+	version := os.Getenv("ORIGIN_VERSION")
```
Comment (bparees) [inline pkg/testframework/versions.go]: this is now set via the job: https://github.com/openshift/aos-cd-jobs/pull/1053

**Comment by bparees** [comment]
@dmage ptal

**Comment by dmage** [comment]
@bparees do you want to merge it now or when the v3.9 tag will be available?

**Comment by bparees** [comment]
@dmage let's hold it until the `origin:v3.9` tag that i've been promised, shows up.

@derekwaynecarr @smarterclayton any chance we can just manually create said tag now(based on what we currently call latest), since we're about to start cutting 3.9 branches and that would enable us to get our CI house in order?

/hold

**Comment by bparees** [comment]
tags (e.g. origin:v3.9) have been created
/hold cancel

**Comment by dmage** [comment]
[flake](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_image-registry/61/test_pull_request_image_registry_extended/201/) openshift/origin#18384
/retest

**Pull request merged**